gobblin-compaction/src/main/java/gobblin/compaction/dataset/DatasetMetric.java

```diff
@@ -0,0 +1,211 @@
+package gobblin.compaction.dataset;
+
+import java.io.IOException;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.commons.lang.StringUtils;
+import org.apache.hadoop.fs.FileStatus;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.fs.PathFilter;
+import org.joda.time.DateTimeZone;
+import org.joda.time.DateTime;
+import org.joda.time.Period;
+import org.joda.time.format.PeriodFormatter;
+import org.joda.time.format.PeriodFormatterBuilder;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.base.Splitter;
+import com.google.common.collect.Lists;
+import com.google.common.collect.Maps;
+
+import gobblin.compaction.conditions.RecompactionCondition;
+import gobblin.compaction.conditions.MultiRecompactionCondition;
+import gobblin.compaction.conditions.RecompactionConditionBasedOnFileCount;
+import gobblin.compaction.conditions.RecompactionConditionBasedOnPercent;
+import gobblin.compaction.conditions.RecompactionConditionBasedOnTime;
+import gobblin.compaction.mapreduce.MRCompactor;
+import gobblin.util.DatasetFilterUtils;
+import gobblin.util.FileListUtils;
+import gobblin.util.RecordCountProvider;
+import gobblin.util.recordcount.LateFileRecordCountProvider;
+
+/**
+ * A class {@link DatasetMetric} which provides runtime metrics for a given dataset.
+ *
+ * The class also contains different recompaction criteria {@link RecompactionCondition}, which indicates if a
+ * recompaction is needed. These conditions will be examined by {@link gobblin.compaction.mapreduce.MRCompactorJobRunner}
+ * after late data was found and moved from inputDir to outputLateDir.
+ */
+
+public class DatasetMetric {
```

Comment:
Is there any reason to call this class `Metric`? This is more like a helper class for recompaction.